packages/block-library/src/site-title/block.json

```diff
@@ -75,5 +75,8 @@
 		}
 	},
 	"editorStyle": "wp-block-site-title-editor",
-	"style": "wp-block-site-title"
+	"style": "wp-block-site-title",
+	"selectors": {
+		"typography": ".wp-block-site-title > span, .wp-block-site-title > a"
```

Comment:
This introduces a regression where the editor and frontend apply different styles. The Site Title block doesn't render an inner `span` on the frontend.

As [proposed](https://github.com/WordPress/gutenberg/issues/65169#issuecomment-2345321749) over on https://github.com/WordPress/gutenberg/issues/65169, I have some ideas to address this and the other issue flagged in https://github.com/WordPress/gutenberg/pull/64911#issuecomment-2339018022.

I'll get a draft PR up soon.